### PR TITLE
Update for using 'master' as primary development branch

### PR DESCRIPTION
**Pull request by jeffaco**
@Microsoft/ostc-devs @Microsoft/omsagent-devs

### CODE REVIEW

**Comment by niroyb** [comment]
Signed off

File: BRANCHING.md [hunk 1]
```diff
@@ -6,18 +6,23 @@ for code reviews. These branches are not the focus of this document, as they
 are generally short lived (as short as a few minutes, or as long as it takes
 for a developer to complete his work).
 
-When working on branch “develop” (equivalent of the improvement branch on Team
+When working on branch “master” (equivalent of the improvement branch on Team
 Foundation Server, our prior source code management system):
 
-1. We will “normally” have one development stream: Branch “develop”. This is
+1. We will normally have one development stream: Branch “master”. This is
 the equivalent of the “improvement” branch in TFS, and represents our
```
Comment (robbiezhang) [inline BRANCHING.md]: the equivalent of the "stable" branch?

File: BRANCHING.md [hunk 2]
```diff
@@ -6,18 +6,23 @@ for code reviews. These branches are not the focus of this document, as they
...
 “latest/greatest” code. Most developers work on this branch, or a sub-branch
 of this branch, the majority of the time.
 
-2. We will have a “master” branch; we will merge to this when we want to test
-and stabilize for a release (if needed). Note that the “master” branch will
-generally be the last “stable” release of the “develop” branch; no other
-branches should be merged there. Once stabilized, we'll tag it for release.
+2. When we need a branch to stabilize a release, we will create a branch on the
```
Comment (robbiezhang) [inline BRANCHING.md]: do you mean "create a branch" on the super-project or the individual repository?

File: BRANCHING.md (hunk 1 above)
Comment (jeffaco) [inline BRANCHING.md]: Well, the intent is that we never break master. We test to be sure our changes are good prior to checkin. If master is broken, we fix it immediately (with priority) as to not impact other developers.

But that said, calling it "stable" without more testing is a stretch. We used to have PBUILD that gave us a lot more confidence across many systems, but that no longer works due to git. We do have plans to bring that back onboard, but we haven't gotten to that yet.

I guess, to summarize: Master "should" work, but if it doesn't, we quickly fix it as soon as we find out about it. After a big merge to Master, things are more likely to be touchy for a few days.

File: BRANCHING.md (hunk 2 above)
Comment (jeffaco) [inline BRANCHING.md]: The super-project has no code to speak of, so there's really nothing to stabilize. When I say "create a branch", I mean create a branch on the individual repository so we can fix the code.

Branching in the super-project is only necessary if we need to tag something that's different in some way. For example, I needed to do this yesterday. Some of the repositories were rewritten to eliminate large files that should never have been committed. This caused a rewrite of the complete history. In order to tag for 1.0.0-21, I needed to create a temporary branch, get the commit hashs for the subprojects correct, commit that, tag it, and then delete the branch.

That way, the tag of v1.0.0-21 actually points to the code that went into v1.0.0-21.

In general, I don't anticipate that we'd ever do this "normally" in the super-project, unless we were doing a hotfix or something building against specific branches of subprojects.

**Pull request merged**